src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/prison/UserServiceTest.kt

```diff
@@ -6,85 +6,164 @@ import org.junit.jupiter.api.Nested
 
 class UserServiceTest {
-  private val nomisUserApiService: UserApiService = mock()
+
+  private val prisonUserApiService: UserApiService = mock()
   private val authApiService: AuthApiService = mock()
   private val notificationService: NotificationService = mock()
   private val verifyEmailDomainService: VerifyEmailDomainService = mock()
-  private val nomisUserService = UserService(
-    nomisUserApiService,
+  private val verifyEmailService: VerifyEmailService = mock()
+  private val prisonUserService = UserService(
+    prisonUserApiService,
     authApiService,
     notificationService,
     verifyEmailDomainService,
+    verifyEmailService,
   )
 
+  @Nested
+  inner class ChangeEmail {
+    private val userName = "testy"
+    private val newEmailAddress = "new.testy@testing.com"
+
+    @Test
+    fun `should throw exception when prison user not present in prison system`() {
+      whenever(prisonUserApiService.findUserByUsername(userName)).thenReturn(null)
+
+      assertThatThrownBy { prisonUserService.changeEmail(userName, newEmailAddress) }
+        .isInstanceOf(EntityNotFoundException::class.java)
+        .hasMessage("Prison username $userName not found")
+
+      verify(authApiService, never()).confirmRecognised(any())
+      verify(verifyEmailService, never()).requestVerification(any(), any())
+      verify(authApiService, never()).updateEmail(any(), any())
+    }
+
+    @Test
+    fun `should throw exception when user not recognised by Auth`() {
+      whenever(prisonUserApiService.findUserByUsername(userName)).thenReturn(createNomisUserDetails())
+      doThrow(RuntimeException("Auth API call failed")).whenever(authApiService).confirmRecognised(userName)
+
+      assertThatThrownBy { prisonUserService.changeEmail(userName, newEmailAddress) }
+        .isInstanceOf(RuntimeException::class.java)
+        .hasMessage("Auth API call failed")
+
+      verify(verifyEmailService, never()).requestVerification(any(), any())
+      verify(authApiService, never()).updateEmail(any(), any())
+    }
+
+    @Test
+    fun `should request verification of new email address`() {
+      val prisonUser = createNomisUserDetails()
+      whenever(prisonUserApiService.findUserByUsername(userName)).thenReturn(prisonUser)
+      whenever(verifyEmailService.requestVerification(prisonUser, newEmailAddress)).thenReturn(
+        VerifyEmailService.LinkEmailAndUsername("link", newEmailAddress, userName),
+      )
+
+      prisonUserService.changeEmail(userName, newEmailAddress)
+
+      verify(verifyEmailService).requestVerification(prisonUser, newEmailAddress)
+    }
+
+    @Test
+    fun `should update email address in Auth`() {
+      val prisonUser = createNomisUserDetails()
+      whenever(prisonUserApiService.findUserByUsername(userName)).thenReturn(prisonUser)
+      whenever(verifyEmailService.requestVerification(prisonUser, newEmailAddress)).thenReturn(
+        VerifyEmailService.LinkEmailAndUsername("link", newEmailAddress, userName),
```

Comment:
can this be imported?

Reply:
Had been defined as a nested data class (rightly or wrongly), so needed to be qualified with containing class name. I've made it (plus another exception) top level classes (defined in same file)